tests/e2e/csi_cns_telemetry_statefulsets.go

```diff
@@ -0,0 +1,172 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/onsi/ginkgo"
+	"github.com/onsi/gomega"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/test/e2e/framework"
+	fss "k8s.io/kubernetes/test/e2e/framework/statefulset"
+)
+
+/*
+	Test performs following operations
+
+	Steps
+	1. Create a storage class.
+	2. Create nginx service.
+	3. Create nginx statefulsets with 3 replicas.
+	4. Wait until all Pods are ready and PVCs are bounded with PV.
+	5. Expect PVC to pass and cluster distribution value set to latest update
+	6. Scaledown nginx statefulsets with 0 replicas.
+	7. Delete all PVCs from the tests namespace.
+	8. Delete the storage class.
+*/
+
+var _ = ginkgo.Describe("[csi-block-vanilla] [csi-file-vanilla] [csi-supervisor] [csi-guest] CNS-CSI Cluster Distribution for StatefulSets", func() {
+	f := framework.NewDefaultFramework("csi-cns-telemetry")
+	var (
+		namespace         string
+		client            clientset.Interface
+		storagePolicyName string
+		scParameters      map[string]string
+	)
+	ginkgo.BeforeEach(func() {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		namespace = getNamespaceToRunTests(f)
+		client = f.ClientSet
+		bootstrap()
+		sc, err := client.StorageV1().StorageClasses().Get(ctx, storageclassname, metav1.GetOptions{})
+		if err == nil && sc != nil {
+			gomega.Expect(client.StorageV1().StorageClasses().Delete(ctx, sc.Name, *metav1.NewDeleteOptions(0))).NotTo(gomega.HaveOccurred())
+		}
+		scParameters = make(map[string]string)
+		storagePolicyName = GetAndExpectStringEnvVar(envStoragePolicyNameForSharedDatastores)
+		if vanillaCluster {
+			//Reset the cluster distribution value to default value "CSI-Vanilla"
+			setClusterDistribution(ctx, client, vanillaClusterDistribution)
+		}
+	})
+
+	ginkgo.AfterEach(func() {
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		if supervisorCluster {
+			deleteResourceQuota(client, namespace)
+		}
+		ginkgo.By(fmt.Sprintf("Deleting all statefulsets in namespace: %v", namespace))
+		fss.DeleteAllStatefulSets(client, namespace)
+		ginkgo.By(fmt.Sprintf("Deleting service nginx in namespace: %v", namespace))
+		err := client.CoreV1().Services(namespace).Delete(ctx, servicename, *metav1.NewDeleteOptions(0))
+		if !apierrors.IsNotFound(err) {
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}
+
+		if vanillaCluster {
```

Comment:
Should we need to reset cluster distribution value in before and after each blocks?

Reply:
Yes, to make sure that while running on any setup - the before value is the desired value and after running the test the value is reset to the original value.